pkg/translator/jaeger/jaegerproto_to_traces.go

```diff
@@ -252,30 +270,55 @@ func setInternalSpanStatus(attrs pdata.AttributeMap, dest pdata.SpanStatus) {
 	}
 }
 
-func getStatusCodeValFromAttr(attrVal pdata.AttributeValue) (int, error) {
-	var codeVal int64
+// extractStatusDescFromAttr returns the OTel status description from attrs
+// along with true if it is set. Otherwise, an empty string and false are
+// returned. The OTel status description attribute is deleted from attrs in
+// the process.
+func extractStatusDescFromAttr(attrs pdata.AttributeMap) (string, bool) {
+	if msgAttr, ok := attrs.Get(conventions.OtelStatusDescription); ok {
+		msg := msgAttr.StringVal()
+		attrs.Delete(conventions.OtelStatusDescription)
+		return msg, true
+	}
+	return "", false
+}
+
+// Allow tests to check int range errors.
+var validIntRange = func(val int64) error {
+	if val > maxInt || val < minInt {
+		return fmt.Errorf("%w: %d", errRange, val)
+	}
+	return nil
+}
```

Comment:
Where does this requirement with "int" comes from? It is extremely bad design to validate based on the arch type (32/64). It means that if you are running a 32bit version of the collector for the same received data you may get errors that you don't get on a 64bit version of the collector.

Reply:
Sure, I'll revert to just validating int32. :+1: